### PR TITLE
maintainer success factor baseline deviation

### PR DESCRIPTION
**Pull request by bjormel**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/BcpGroupUpdater.java [hunk 1]
```diff
@@ -75,12 +75,12 @@ protected double maintain() {
...
             }
         }
-        double successFactor = asSuccessFactor(attempts, failures);
-        if ( successFactor == 0 )
+        double successFactorDeviation = asSuccessFactorDeviation(attempts, failures);
+        if ( successFactorDeviation == 1.0 )
             log.log(Level.WARNING, "Could not update traffic share on any applications", lastException);
-        else if ( successFactor < 0.9 )
+        else if ( successFactorDeviation > 0.1 )
             log.log(Level.FINE, "Could not update traffic share on all applications", lastException);
```
Comment (olaaun) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/BcpGroupUpdater.java:82]: `successFactorDeviation` is in the range [-baseline, 1 - baseline]

**Review by olaaun — CHANGES_REQUESTED** [review]

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/BcpGroupUpdater.java (hunk 1 above)
Comment (bjormel) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/BcpGroupUpdater.java:82]: @olaaun , question is if I also should set deviation to 0, when there has been no attempts, or if it makes sense to distinguish this case?
```
    protected final double asSuccessFactorDeviation(int attempts, int failures) {
        double factor = attempts == 0 ? 1.0 : 1 - (double)failures / attempts;
        return new BigDecimal(factor - successFactorBaseline).setScale(2, RoundingMode.HALF_UP).doubleValue();
    }
```

Comment (olaaun) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/BcpGroupUpdater.java]: This still applies. Baseline is set to .7, so deviation can never be -1

Comment (bjormel) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/BcpGroupUpdater.java]: :(
Waiting for a brain reboot

**Review by olaaun — APPROVED** [review]

**Pull request merged**